src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs

```diff
@@ -28,14 +28,28 @@ public ServiceHubContext(IHubContext<Hub> hubContext, IHubLifetimeManagerForUser
 
         public void Dispose()
         {
+            if (GetServiceConnectionStatus() != ServiceConnectionStatus.Disconnected)
+            {
+                StopConnectionAsync().Wait();
```

Comment:
No please. Never use .Wait or .Result.

Reply:
OK. How about `_ = StopConnectionAsync()`

Reply:
Nope, if you need to do asynchronous work, then await the result. Remove the IDisposable implementation and add a DisposeAsync that callers can use.

Reply:
OK. I will remove the interface `IDisposable` and its implementation and keep `DisposeAsync`.